openlibrary/macros/LoanStatus.html

```diff
@@ -18,8 +18,9 @@
 $ waiting_loan = check_loan_status and ocaid and ctx.user and ctx.user.get_waiting_loan_for(ocaid)
 $ my_turn_to_borrow = waiting_loan and waiting_loan['status'] == 'available' and waiting_loan['position'] == 1
 
-$if not availability and allow_expensive_availability_check and ocaid:
-  $ availability.update(get_cached_groundtruth_availability(ocaid))
+$# Replace existing, possibly inaccurate, availability with results ground truth availability API call: 
+$if allow_expensive_availability_check and ocaid:
+  $ availability = get_cached_groundtruth_availability(ocaid)
```

Comment:
I chose to reassign the value of `availability` instead of using `update` for two reasons:
1. If `availability` was populated with data from an earlier `get_availability` call, an update would new fields from the ground truth availability API call to the existing dictionary.  My concern is that this would cause unintended side effects somewhere else in the code.
2. If the update happens on a work page, it also updates the availability of the first edition in the editions table.  While this has the benefit of keeping the buttons in sync between components, it masks the underlying data sync issue that is actually causing the buttons to mismatch in most of the cases that I've seen.

Reply:
This clobbers the `identifier` and `collection` field (at least). Agreed it would be nice if we could assign to result of either bulk / ground_truth API and reliably get the same fields back, but not yet the case :(

```suggestion
  $ availability.update(get_cached_groundtruth_availability(ocaid))
```